### PR TITLE
gh-115986: Use param list to mark up pprint.PrettyPrinter constructor

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-115986

----
📚 Documentation preview 📚: https://cpython-previews--116085.org.readthedocs.build/en/116085/library/pprint.html

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @Privat33r-dev

**Comment by Privat33r-dev** [comment]
Side note:
> The configuration parameters stream, indent, width, depth, compact, sort_dicts and underscore_numbers are passed to the [PrettyPrinter](https://cpython-previews--116085.org.readthedocs.build/en/116085/library/pprint.html#pprint.PrettyPrinter) constructor and their meanings are as described in its documentation **above**.
Just noticed + 1 more 'above' mention, which is inaccurate after we moved class to the bottom.

File: Doc/library/pprint.rst [hunk 1]
```diff
@@ -117,51 +117,50 @@ Functions
...
+   :param bool compact:
+      Control the way long :term:`sequences <sequence>` are formatted.
+      If ``False`` (the default),
+      each item of a sequence will be formatted on a separate line,
+      otherwise as many items as will fit within the *width*
+      will be formatted on each output line.
+
+   :param bool sort_dicts:
+      If ``True`` (the default), dictionaries will be formatted with
+      their keys sorted, otherwise they will display in insertion order.
```
Comment (Privat33r-dev) [inline Doc/library/pprint.rst:158]: Maybe a passive tense would be more accurate choice? `will be displayed`. Though I am not sure.

**Comment by erlend-aasland** [comment]
> Just noticed 2 documentation above mentions, which is inaccurate after we moved class to the bottom.

Good catch. Can you please open a PR to fix that?

**Review by AlexWaygood — COMMENTED** [review]
I think the fundamental problem with this module's docs is that they've been written by somebody thinking about it from the perspective of how the module is implemented, rather than from the perspective of how users generally use it. From the perspective of the implementation, it makes sense to document the meaning of all these parameters as part of the PrettyPrinter class, since all the module-level functions are just abstractions over that class. But from the perspective of the user, I'm not sure that really makes sense — users of this module rarely need to use the low-level class.

So maybe the meaning of these parameters should be documented in a  separate table somewhere in these docs, rather than as part of the documentation of this class — we could just say the parameters have the same meaning for all functions in the module

**Comment by Privat33r-dev** [comment]
> Good catch. Can you please open a PR to fix that?

#116104

**Comment by erlend-aasland** [comment]
I agree, @AlexWaygood. We ran into similar problems in the ftplib docs. Perhaps we should consider this approach over there as well.

I'll close this PR for now.

**Comment by Privat33r-dev** [comment]
> > Is there an example of this style?
> 
> I'm imagining a table that looks a bit like https://docs.python.org/3/library/typing.html#id7 (source code at
> 
> https://github.com/python/cpython/blob/86e5e063aba76a7f4fc58f7d06b17b0a4730fd8e/Doc/library/typing.rst#L2736
> 
> )

I wish there was a more modern design for the table though. I am not sure where I can post proposal for it, but it would be nice if instead of this:
image

It looked at least like this:
image

**Comment by erlend-aasland** [comment]
@Privat33r-dev, I'd try and ask in the Documentation category on Discourse, or on our Python Docs Discord server.

**Pull request closed without merging**